<?xml version="1.0" encoding="UTF-8"?>

<launch>
  <!-- Command-line arguments for remote machine, defaults to shrimp opts -->
  <arg name="user" default="shrimp"/>
  <arg name="ip" default="192.168.50.81"/>

  <!-- Load config and parameters -->
  <rosparam file="$(find controller)/configs/$(arg user)_config.yaml" command="load"/>
  <rosparam file="$(find controller)/params.yaml" command="load"/>
  <rosparam file="$(find controller)/configs/motion_planner_config.yaml" command="load"/>

  <!-- Launch logger -->
  <node name="roslogger" pkg="controller" type="roslogger.py" output="screen"/>

  <!-- Launch nodes remotely, assuming there is an /env/user_env.sh script -->
  <machine name="lean" address="$(arg ip)" user="$(arg user)" env-loader="/home/$(arg user)/lean_ws/src/controller/env/$(arg user)_env.sh"/>
  <node machine="lean" name="map_spawner" pkg="controller" type="spawn_map.py" output="screen"/>
</launch>
